portal-api/src/v1/controllers/mandatoryCriteria.controller.js

```diff
@@ -98,14 +99,31 @@ exports.update = async (req, res) => {
 };
 
 exports.delete = async (req, res) => {
-  const collection = await db.getCollection('mandatoryCriteria');
-  const { version } = req.params;
-  const versionNumber = Number(version);
+  const versionNumber = Number(req.params.version);
+  const auditDetails = generatePortalAuditDetails(req.user._id);
 
-  if (!Number.isNaN(versionNumber)) {
-    const status = await collection.deleteOne({ version: { $eq: versionNumber } });
-    return res.status(200).send(status);
+  if (Number.isNaN(versionNumber)) {
+    return res.status(400).send({ status: 400, message: 'Invalid mandatory criteria version number' });
   }
 
-  return res.status(400).send({ status: 400, message: 'Invalid mandatory criteria version number' });
+  const collection = await db.getCollection(MONGO_DB_COLLECTIONS.MANDATORY_CRITERIA);
+  const mandatoryCriteria = await collection.findOne({ version: { $eq: versionNumber } }, { projection: { _id: true } });
+
+  if (!mandatoryCriteria) {
+    return res.status(404).send({ status: 404, message: 'Mandatory Criteria not found' });
+  }
+
+  try {
+    const deleteResult = await deleteOne({
+      documentId: new ObjectId(mandatoryCriteria._id),
+      collectionName: MONGO_DB_COLLECTIONS.MANDATORY_CRITERIA,
+      db,
+      auditDetails,
+    });
+
+    return res.status(200).send(deleteResult);
+  } catch (error) {
+    console.error('Error occurred deleting mandatory criteria, %o', error);
```

Comment:
Im generally not keen on adding error handling here, as we should have enough just by the errors automatically logging, but happy to keep as it follows the pattern

Reply:
This is needed to test the transactions, otherwise they just throw & fail the test